app/src/lib/editors/linux.ts

```diff
@@ -51,6 +56,22 @@ function getEditorPath(editor: ExternalEditor): Promise<string | null> {
       return getPathIfAvailable('/usr/bin/subl')
     case ExternalEditor.Typora:
       return getPathIfAvailable('/usr/bin/typora')
+    case ExternalEditor.SlickEdit:
+      const possiblePaths = [
+        '/opt/slickedit-pro2018/bin/vs',
+        '/opt/slickedit-pro2017/bin/vs',
+        '/opt/slickedit-pro2016/bin/vs',
+        '/opt/slickedit-pro2015/bin/vs',
+      ]
+      for (const possiblePath of possiblePaths) {
+        const slickeditPath = getPathIfAvailable(possiblePath)
+        if (slickeditPath) {
+          return slickeditPath
+        }
+      }
+      return new Promise<string | null>((resolve, reject) => {
```

Comment:
I think if you make this a `async function getEditorPath()` you can use `await` in here to make things more readable. There's a gotcha with `getPathIfAvailable` in that it returns a `Promise<string | null>`, and your check `if (slickeditPath)` is only checking if the `Promise` is not null (which is always is), rather than the value returned from the promise. 

I think tweaking the flow like this should help with what you're trying to do here:

```ts
    case ExternalEditor.SlickEdit:
      const possiblePaths = [
        '/opt/slickedit-pro2018/bin/vs',
        '/opt/slickedit-pro2017/bin/vs',
        '/opt/slickedit-pro2016/bin/vs',
        '/opt/slickedit-pro2015/bin/vs',
      ]
      for (const possiblePath of possiblePaths) {
        const slickeditPath = await getPathIfAvailable(possiblePath)
        if (slickeditPath) {
          return slickeditPath
        }
      }
      return null
```

Reply:
Yes, I think that is much better and doing what I expected.

Thank you very much.